src/sources/util/tcp/mod.rs

```diff
@@ -334,6 +350,17 @@ async fn handle_stream<T>(
                             }
                         }
 
+                        if let Some(tls_peer_key) = &tls_peer_key {
+                            if let Some(certificate_metadata) = &certificate_metadata {
+                                let mut metadata: BTreeMap<String, value::Value> = BTreeMap::new();
+                                metadata.insert("certificate".to_string(), certificate_metadata.to_string().into());
```

Comment:
Could we call this `subject`?
```suggestion
                                metadata.insert("subject".to_string(), certificate_metadata.to_string().into());
```